subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/install/internal/JavaToolchainRepository.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.jvm.toolchain.install.internal;
+
+import org.gradle.jvm.toolchain.JavaToolchainSpec;
+
+import java.net.URI;
+
+public interface JavaToolchainRepository {
+
+    boolean canProvide(JavaToolchainSpec spec);
```

Comment:
I do not think this is needed with this contract. In the past, checking first there before doing the download makes sense. But now we can use the `toUri` directly.

However we should have some method, potentially with a default implementation, that indicates with which version of the `JavaToolchainSpec` there is compatibility:
`SomeEnum getToolchainSpecCompatibility()`